### DISCUSSION TITLE
Inverted Colors after Upgrading for .tiff

### DISCUSSION THREAD

**Discussion by trouaux** [ImageMagick/ImageMagick#2935] [Help]
Hi IM community, 

I ran into an issue when converting a .tif file, after looking at open issues I found #2873. My issue is the same but it should have been fixed in the latest release but I still have the issue for my file, could you tell me if it has been patched in the latest release or it's a particular issue to my file ?

```
Version: ImageMagick 7.0.10-46 Q16 x64 2020-12-05 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 192829334
Features: Cipher DPC OpenCL
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib
```

Source: https://www.dropbox.com/s/4tr56pekinu2qan/7290107646376.tif?dl=0

Regards,
Thomas

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Reply by Jmuccigr** [reply]
I think I'm seeing this or something like it. Doing simple actions on a tiff has odd results. For example on a Group4 formatted file, it negates the colors:

`magick aaa.tiff -define tiff:preserve-compression=true abb.tiff`

(Same visual result with or without the -define.)

`magick rose: rose.tiff; magick rose.tiff rose2.tiff`

gives strange results. All attached.

[Archive.zip](https://github.com/ImageMagick/ImageMagick/files/5713109/Archive.zip)

MacOS 10.15.7

Version: ImageMagick 7.0.10-48 Q16 x86_64 2020-12-16 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib fftw fontconfig freetype heic jng jp2 jpeg lcms ltdl lzma png tiff webp wmf x xml zlib

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Reply by Jmuccigr** [reply]
Thanks. This seems to have done it. Version 7.0.10-50. (Just updated to 7.0.10-51.)
